src/components/scenes/Loans/LoanCreateScene.js

```diff
@@ -46,36 +47,51 @@ type Props = {
 export const LoanCreateScene = (props: Props) => {
   const { navigation, route } = props
   const { borrowEngine, borrowPlugin } = route.params
-  const { currencyWallet: srcWallet, debts } = borrowEngine
+  const { currencyWallet: beWallet, debts } = borrowEngine
 
   const theme = useTheme()
   const styles = getStyles(theme)
 
+  // Skip directly to LoanStatusScene if an action for the same actionOpType is already being processed
   const existingProgramId = useRunningActionQueueId('loan-create', beWallet.id)
+  if (existingProgramId != null) navigation.navigate('loanStatus', { actionQueueId: existingProgramId })
+
   if (debts.length > 0) {
     // TODO: transition to "advanced" loan details scene
   }
 
   // Wallet/Token Data
   const account = useSelector(state => state.core.account)
   const wallets = useWatch(account, 'currencyWallets')
-  const { fiatCurrencyCode: isoFiatCurrencyCode, id: srcWalletId } = srcWallet
-  const [srcTokenId, setSrcTokenId] = useState()
-  const [srcCurrencyCode, setSrcCurrencyCode] = useState('')
-  const fiatCurrencyCode = isoFiatCurrencyCode.replace('iso:', '')
   const allTokens = useAllTokens(account)
 
-  const srcPluginId = srcWallet.currencyInfo.pluginId
-  const srcWalletName = useWalletName(srcWallet)
-  const srcBalance = useWalletBalance(srcWallet, srcTokenId ?? undefined)
-  const srcToken = useMemo(() => (srcTokenId != null ? allTokens[srcPluginId][srcTokenId] : {}), [allTokens, srcPluginId, srcTokenId])
-  const srcAssetName = useMemo(() => (srcToken != null ? srcToken.displayName : ''), [srcToken])
+  const { fiatCurrencyCode: isoFiatCurrencyCode, currencyInfo: beCurrencyInfo } = beWallet
+  const fiatCurrencyCode = isoFiatCurrencyCode.replace('iso:', '')
+  const bePluginId = beCurrencyInfo.pluginId
+
+  // Source Wallet Data
+  const [srcWalletId, setSrcWalletId] = useState()
+  const [srcTokenId, setSrcTokenId] = useState()
+  const [srcCurrencyCode, setSrcCurrencyCode] = useState()
+
+  const srcWallet = srcWalletId == null ? null : wallets[srcWalletId]
+  const srcPluginId = srcWallet == null ? null : srcWallet.currencyInfo.pluginId
+  const srcToken = useMemo(() => (srcTokenId != null && srcPluginId != null ? allTokens[srcPluginId][srcTokenId] : {}), [allTokens, srcPluginId, srcTokenId])
+  const srcBalance = useWalletBalance(srcWallet ?? beWallet, srcTokenId) // HACK: Balance isn't being used anyway if the src wallet hasn't been chosen yet. Default to the borrow engine wallet in this case so this hook can be used
```

Comment:
This abstraction failed us... We need a `useMaybeWalletBalance` abstraction now.

Reply:
Let's discuss, I think this is causing the bugs around auto wallet creation/enabling tokens during the gui loan workflow

Reply:
We'll need one for the useWalletName too, let's punt this for another PR.

https://app.asana.com/0/1200382638405084/1202815989420272/f